### PR TITLE
Space id integration

### PR DESCRIPTION
**Pull request by mishramonalisha76**
### Ticket Number

- [1575](https://github.com/push-protocol/push-dapp/issues/1575)

### Description

- **Problem/Feature**: Enhanced the domain resolution process by using the space-id package. Now all domain supported by space-id can be resolved, including bnb , eth and unstoppable domains.

### Type of Change

- [ ] Bug fix
- [x] New feature
- [ ] Code refactor
- [ ] Documentation update
- [ ] Other (please describe):

### Checklist

- [x] **Quick PR**: Is this a quick PR? Can be approved before finishing a coffee.
  - [x] Quick PR label added
- [x] **Not Merge Ready**: Is this PR dependent on some other PR/tasks and not ready to be merged right now.
  - [x] DO NOT Merge PR label added

### Frontend Guidelines

- [x] Followed frontend guidelines as per the [Guidelines Notion Doc](https://www.notion.so/pushprotocol/Frontend-dApp-Guidelines-1d7806ae3d9e4569a340b563dcd0536c)

### Build & Testing

- [x] No errors in the build terminal
- [x] Engineer has tested the changes on their local environment
- [x] Engineer has tested the changes on deploy preview

### Screenshots/Video with Explanation

- **Before:** Explain the previous behavior

- **After:** What's changed now

### Review & Approvals

- [ ] Self-review completed
- [ ] Code review by at least one other engineer
- [ ] Documentation updates if applicable

### CODE REVIEW

**Comment by rohitmalhotra1420** [comment]
@mishramonalisha76 reopened this pull request.

File: package.json [hunk 1]
```diff
@@ -44,6 +44,7 @@
     "@testing-library/user-event": "^7.1.2",
     "@uniswap/widgets": "^2.47.3",
     "@unstoppabledomains/resolution": "8.5.0",
+    "@web3-name-sdk/core": "^0.1.18",
     "@web3-onboard/coinbase": "^2.2.5",
```
Comment (rohitmalhotra1420) [inline package.json:48]: @mishramonalisha76 the ticket description says resolved the domain name using space id package, but where is the space id package used?

File: src/hooks/useResolveWeb3Name.ts [hunk 2]
```diff
@@ -13,21 +14,23 @@ import { AppContextType } from 'types/context';
...
-  provider.lookupAddress(checksumWallet).then(async (ens) => {
-    if (ens) {
-      ensName = ens;
-      setWeb3NameList((prev) => ({ ...prev, [checksumWallet]: ens }));
-    } else {
-      ensName = null;
-    }
-  });
-  return ensName;
+const getDomainName = async (checksumWallet: string, setWeb3NameList: any) => {
```
Comment (rohitmalhotra1420) [inline src/hooks/useResolveWeb3Name.ts:17]: If we have this functionality inside the sdk library then why not use it from there and avoid installing the space id library here?

**Review by rohitmalhotra1420 — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (mishramonalisha76) [reply package.json:48]: "@web3-name-sdk/core" this the space id package

File: src/hooks/useResolveWeb3Name.ts (hunk 2 above)
Comment (mishramonalisha76) [reply src/hooks/useResolveWeb3Name.ts:17]: not sure why its not getting installed from the sdk itself, even though its mentioned in dependencies in uiweb package

**Review by rohitmalhotra1420 — DISMISSED** [review]
@mishramonalisha76 this needs a second approval from @corlard3y .

**Review by corlard3y — APPROVED** [review]

**Review by rohitmalhotra1420 — APPROVED** [review]

**Pull request merged**